server/app/services/applications/ProgramAdminApplicationService.java

```diff
@@ -58,4 +64,20 @@ public void setStatus(Application application, StatusEvent newStatus, Account ad
             application, admin, ApplicationEventDetails.Type.STATUS_CHANGE, details);
     eventRepository.insertSync(event);
   }
+
+  /**
+   * Saves a new {@link ApplicationEventDetails.Type.NOTE_CHANGE} event.
+   *
+   * @param admin The Account that instigated the change.
+   */
+  public void setNote(Application application, NoteEvent note, Account admin) {
```

Comment:
Not blocking: Reading the docstring next to the method name is a bit incongrous (docstring=Saves a new and methodname=setNote). setNote implies there's a single note, while we model it as a new event of a given type. The UI will only display the latest note (at least as I understand). That said, at the service level, it feels like we're tying the method name to that implementation a little too strongly.

Reply:
At some layer we should be imposing business logic on top of the data and I think that's appropriate in the Service layer as it's the lowest non database oriented layer.

There is logically 1 active status or note, there are also historical ones that the Service may later reveal to the a Controller.

As you've brought up before, we don't have great lines as such, often using the data class directly in controllers, but I don't think that is a good thing to not try to impost the business logic of viewing the database in a Service.

Thoughts?

Reply:
Hmm, that's a good approach, at least we can _try_ to enforce business logic through methods and their docs.

I'd prefer to do that in a separate PR though.